### ISSUE TITLE
redocly/cli lint hangs for a while before exiting

### ISSUE THREAD

**Issue by philipnbbc** [Redocly/redocly-cli#1146]
**Describe the bug**

Running version >= `v1.0.0-beta.127` (`latest` is `v1.0.0-beta.129`) of `redocly/cli` Docker results in the validation completing and then the process sits there for a while before exiting. The wait time can be significant (30 sec) for larger OpenAPI spec files.

**To Reproduce**

A small OpenAPI file, `example.yaml`:
```yaml
openapi: 3.0.3
info:
  title: Test
paths:
  /:
    get:
      responses:
        '200':
```

1. Given the OpenAPI example file shown above
2. Run `docker run --rm -v $PWD:/spec redocly/cli lint example.yaml`
3. It outputs "run `redocly lint --generate-ignore-file` to add all problems to the ignore file." and then sits there for a couple seconds
4. Run `docker run --rm -v $PWD:/spec redocly/cli:v1.0.0-beta.126 lint example.yaml` and it exits immediately

Using a larger OpenAPI file results in a longer and more significant hang time.

**Expected behavior**

It shouldn't, or it is unclear why it would hang for a while before exiting.

**Redocly Version(s)**

Issue is with Docker image `redocly/cli` version >= `v1.0.0-beta.127` (`latest` == `v1.0.0-beta.129`)

**Comment by philipnbbc** [comment]
Tried something... It turns out it hangs when behind a proxy. I assume it is attempting to make a HTTP request?

Can that be behavior be disabled? Switching telemetry off doesn't make a difference: `docker run --rm -e REDOCLY_TELEMETRY=off -v $PWD:/spec redocly/cli lint example.yaml`

**Comment by lornajane** [comment]
Thanks for the update, although you've disproved my working hypothesis which was that the telemetry could cause a problem despite our testing on it! We'll take a look and keep you posted.

**Comment by tatomyr** [comment]
Hi @philipnbbc, 
Could you check if setting env variable `CI=true` solves this?

**Comment by philipnbbc** [comment]
> Could you check if setting env variable `CI=true` solves this?

I ran `docker run --rm -e CI=true -v $PWD:/spec redocly/cli lint example.yaml` and the issue remains.

The output of tcpdump showed DNS queries for `api.redocly.com`.

**Comment by tatomyr** [comment]
Which version is this? Could you also switch off the telemetry? My last guess is this is a combo 😅

**Comment by philipnbbc** [comment]
:tada: That did the trick. It returns immediately and tcpdump no longer shows `api.redocly.com`. This was the `latest` / `v1.0.0-beta.129` version.

**Comment by yuan1238y** [comment]
I tried set the environment `REDOCLY_TELEMETRY=off` resolve this problem.
This version is `v1.0.0-beta.129`

**Comment by tatomyr** [comment]
@philipnbbc, @yuan1238y,

Could you check whether the issue has been resolved in version `1.0.0-beta.130`?

**Comment by philipnbbc** [comment]
It hasn't fixed it for me unfortunately. I ran `docker run --rm -v $PWD:/spec redocly/cli:v1.0.0-beta.130 lint example.yaml` and it can hang a couple seconds or a lot more.

I also tried it with `-e REDOCLY_TELEMETRY=off` or `-e CI=true` (not both) and they both hang. It wasn't clear whether one or the other had improved.

**Comment by tatomyr** [comment]
Thanks @philipnbbc. 

Reopening the issue.

**Comment by tatomyr** [comment]
@philipnbbc Hi, could you check it one more time with **beta.131**?

**Comment by philipnbbc** [comment]
I ran `docker run --rm -v $PWD:/spec redocly/cli:v1.0.0-beta.131 lint example.yaml` and it has the same issue.

I confirmed that without docker that @redocly/cli@1.0.0-beta.130 fixed the delay issue.

I don't see the proxy in the tcpdump output and so I guess the fundamental issue is that there is nodejs code trying to connect that doesn't support proxies and recognise the proxy environment variables.

**Comment by tatomyr** [comment]
Thanks @philipnbbc,

We will continue researching this then.

**Comment by tatomyr** [comment]
I'm trying to replicate this by calling a very slow endpoint, and I cannot see it hanging for too long. How long is the delay for you? Has it been changed in the recent updates? Any additional reproduction steps or info you can provide us with (e.g. OS, nodeJS version &c.)?

**Comment by philipnbbc** [comment]
I think the issue is that we have a proxy and network traffic has to go via that proxy. It appears from the tcpdump that the code is attempting to make connections without going via the proxy. I checked that the proxy environment variables are set in the docker container.

The tcpdump shows DNS queries for `registry.npmjs.org` which returns 12 IP addresses. There is then many rounds of attempts made to connect to those addresses. If I add `-e CI=true` it does the same but for `api.redocly.com` and it attempts to connect to the 3 AWS IP addresses.

I ran the cli docker a couple times and it was taking around 2 min 12 sec fairly consistently. It does vary because the delay was less last week.

I've had issues with other nodejs code that don't support proxies, but in this case we can workaround it by using the 2 environment variables.

**Comment by tatomyr** [comment]
I am putting it on hold for now, as there is a workaround.

**Comment by tatomyr** [comment]
Note: [this](https://github.com/Redocly/redocly-cli/issues/529) is tangentially related.

**Comment by tatomyr** [comment]
JFYI @philipnbbc, hopefully this should have been fixed in v1.16.0. 😅

**Comment by philipnbbc** [comment]
> JFYI @philipnbbc, hopefully this should have been fixed in v1.16.0. 😅

Yes it has, thank you :tada:

**Comment by tatomyr** [comment]
@philipnbbc we're going to remove the `node-fetch` dependency, which may impact how Redocly CLI works behind proxies ([here's the corresponding PR](https://github.com/Redocly/redocly-cli/pull/1848)). There's a test version with these changes (`0.0.0-snapshot.1737556585`) which you can try and check if it breaks anything for you. 

cc @DmitryAnansky @yuan1238y

**Comment by philipnbbc** [comment]
@tatomyr, a bit late, but I gave it a try behind the proxy and it works. Thanks!

**Comment by tatomyr** [comment]
This is great news @philipnbbc! Just for your information, we've released the changes in `v1.28.0`.
